src/app/shared/helpers/data-pager.ts

```diff
@@ -21,19 +21,26 @@ interface PagerOptions<T> {
   onLoadMoreError: (error: Error) => void,
 }
 
+interface PagedData<T> {
+  list: T[],
+  newItems: T[],
+}
+
 export class DataPager<T> {
   private trigger$ = new ReplaySubject<{ reset: boolean }>(1);
 
-  state$ : Observable<FetchState<T[]>> = this.trigger$.pipe(
-    switchScan<{ reset: boolean }, FetchState<T[]>, Observable<FetchState<T[]>>>(
+  private state$ : Observable<FetchState<PagedData<T>>> = this.trigger$.pipe(
```

Comment:
In this file, I over-defined the types to have type-checking helping me checking every steps. Probably some of these types are useless now. (here and the switchscan)